### PR TITLE
Guest code updates for  strate chain

### PR DESCRIPTION
**Pull request by MdTeach**
## Description
* Update to strata chain id in the guest code
* Update EL/CL/Checkpoint to enable deposit/withdrawal assert

### Type of Change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature/Enhancement (non-breaking change which adds functionality or enhances an existing one)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation update
-   [ ] Refactor
-   [ ] New or updated tests

## Checklist

-   [ ] I have performed a self-review of my code.
-   [ ] I have commented my code where necessary.
-   [ ] I have updated the documentation if needed.
-   [ ] My changes do not introduce new warnings.
-   [ ] I have added tests that prove my changes are effective or that my feature works.
-   [ ] New and existing tests pass with my changes.

### CODE REVIEW

File: crates/proof-impl/checkpoint/src/lib.rs
Comment (prajwolrg) [inline crates/proof-impl/checkpoint/src/lib.rs:1]: The types makes sense to be consistent across. Let's create a ticket to track this inconsistency.

File: provers/risc0/guest-cl-stf/src/main.rs
Comment (prajwolrg) [inline provers/risc0/guest-cl-stf/src/main.rs:1]: There should not be all this log in the risc0 guest code. Can this be moved to proof-impl?

**Review by prajwolrg — COMMENTED** [review]
Looks good overall. Some inconsistencies.

File: crates/proof-impl/checkpoint/src/lib.rs
Comment (MdTeach) [reply crates/proof-impl/checkpoint/src/lib.rs:1]: Created: **STR-552**

**Review by prajwolrg — APPROVED** [review]

**Review by prajwolrg — APPROVED** [review]

**Pull request merged**